Material.Icons.Avalonia/MaterialIconExt.cs

```diff
@@ -21,36 +23,91 @@ public MaterialIconExt(MaterialIconKind kind, double iconSize, MaterialIconAnima
+        public IBinding? KindBinding { get; set; }
+
+        /// <summary>
+        /// Gets or sets the size of the icon to display.
+        /// </summary>
         [ConstructorArgument("iconSize")]
-        public double? IconSize { get; set; }
+        public object? IconSize { get; set; }
 
+        /// <summary>
+        /// Gets or sets the icon foreground brush.
+        /// </summary>
         [ConstructorArgument("iconForeground")]
         public IBrush? IconForeground { get; set; }
 
+        /// <summary>
+        /// Gets or sets a binding for the icon foreground. Use this when data binding is required.
+        /// </summary>
+        [ConstructorArgument("iconForegroundBinding")]
+        public IBinding? IconForegroundBinding { get; set; }
+
+        /// <summary>
+        /// Gets or sets the animation to play. Provides IntelliSense autocomplete.
+        /// </summary>
         [ConstructorArgument("animation")]
         public MaterialIconAnimation Animation { get; set; }
 
+        /// <summary>
+        /// Gets or sets a binding for the animation. Use this when data binding is required.
+        /// </summary>
+        [ConstructorArgument("animationBinding")]
+        public IBinding? AnimationBinding { get; set; }
+
+        /// <summary>
+        /// Gets or sets the vertical alignment of the content.
+        /// </summary>
         [ConstructorArgument("verticalAlignment")]
         public VerticalAlignment? VerticalAlignment { get; set; }
 
+        /// <summary>
+        /// Gets or sets the horizontal alignment of the content.
+        /// </summary>
         [ConstructorArgument("horizontalAlignment")]
         public HorizontalAlignment? HorizontalAlignment { get; set; }
 
+        /// <summary>
+        /// Gets or sets the class names to apply to the element.
+        /// </summary>
         [ConstructorArgument("classes")]
         public string? Classes { get; set; }
 
         public override object ProvideValue(IServiceProvider serviceProvider) {
-            var result = new MaterialIcon {
-                Kind = Kind,
-                Animation = Animation
-            };
+            var result = new MaterialIcon();
+
+            // Kind: binding takes precedence
+            if (KindBinding is not null) result.Bind(MaterialIcon.KindProperty, KindBinding);
+            else result.Kind = Kind;
+
+            // Animation: binding takes precedence
+            if (AnimationBinding is not null) result.Bind(MaterialIcon.AnimationProperty, AnimationBinding);
+            else result.Animation = Animation;
+
+            if (IconSize is not null) {
+                switch (IconSize) {
+                    case IBinding binding:
+                        result.Bind(MaterialIcon.IconSizeProperty, binding);
+                        break;
+                    case IConvertible conv:
+                        result.IconSize = conv.ToDouble(System.Globalization.CultureInfo.InvariantCulture);
```

Comment:
Are you sure about IConvertible here?

Reply:
I have debug the code and it always return a string, even if we pass 10 on xmal, so using Convert.ToDouble lead to the "same thing" as IConvertible. Every primitive is using `IConvertible` interface, which is better use than strict use the Convert.To. Even if user manage to pass a casted type, IConvertible got all primitives covered.